hydra-node/src/Hydra/API/Server.hs

```diff
@@ -136,24 +142,33 @@ runAPIServer ::
 runAPIServer host port party tracer history callback responseChannel = do
   traceWith tracer (APIServerStarted port)
   handle onIOException $
-    runServer (show host) (fromIntegral port) $ \pending -> do
-      let path = requestPath $ pendingRequest pending
-      queryParams <- uriQuery <$> mkURIBs path
-      con <- acceptRequest pending
-      chan <- STM.atomically $ dupTChan responseChannel
-      traceWith tracer NewAPIConnection
-
-      -- api client can decide if they want to see the past history of server outputs
-      if shouldNotServeHistory queryParams
-        then forwardGreetingOnly con
-        else forwardHistory con
-
-      -- api client can decide if they want tx's to be displayed as CBOR instead of plain json
-      let txDisplay = decideOnTxDisplay queryParams
-
-      withPingThread con 30 (pure ()) $
-        race_ (receiveInputs con) (sendOutputs chan con txDisplay)
+    let serverSettings =
+          setHost (fromString $ show host) $
+            setPort (fromIntegral port) $
+              setOnException (\_ _ -> pure ()) defaultSettings
```

Comment:
`APIConnectionError` and `APIHandshakeError` are unused now.

Should: Just trace any `SomeException` we get here (so it's not hidden).
```suggestion
              setOnException (\_ e -> traceWith tracer $ APIConnectionError { reason = show e}) defaultSettings
```